COPYING

```diff
@@ -39,245 +39,69 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+the above copyright.  See licenses/channel.h-LICENSE for copyright and
+licensing information.
 
 CVC4 incorporates code from ANTLR3, excluded from the above copyright.
 See http://www.antlr3.org/, and the files src/parser/bounded_token_buffer.h,
 src/parser/bounded_token_buffer.cpp, and src/parser/antlr_input_imports.cpp.
-Their copyright:
-
-  [The "BSD licence"]
-  Copyright (c) 2005-2009 Jim Idle, Temporal Wave LLC
-  http://www.temporal-wave.com
-  http://www.linkedin.com/in/jimidle
-
-  All rights reserved.
+See licenses/antlr3-LICENSE for copyright and licensing information.
 
-  Redistribution and use in source and binary forms, with or without
-  modification, are permitted provided that the following conditions
-  are met:
-  1. Redistributions of source code must retain the above copyright
-     notice, this list of conditions and the following disclaimer.
-  2. Redistributions in binary form must reproduce the above copyright
-     notice, this list of conditions and the following disclaimer in the
-     documentation and/or other materials provided with the distribution.
-  3. The name of the author may not be used to endorse or promote products
-     derived from this software without specific prior written permission.
-
-  THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
-  IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
-  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
-  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
-  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
-  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
-  THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+CVC4 optionally links against the LFSC proof checker. See licenses/LFSC-LICENSE
+for copyright and licensing information.
 
 CVC4 by default links against The GNU Multiple Precision (GMP) Arithmetic
-Library, which is licensed under GNU LGPL v3. See the file GMP-LICENSE for a
-copy of that license.  Note that according to the terms of the LGPL, both
+Library, which is licensed under GNU LGPL v3. See the file licenses/lgpl.txt
+for a copy of that license.  Note that according to the terms of the LGPL, both
 CVC4's source, and the combined work (CVC4 linked with GMP) may (and do) remain
 under the more permissive CVC4 license.
 
-Alternatively, CVC4 can be optionally configured to link against CLN, the Class
-Library for Numbers, available here:
+-------------------------------------------------------------------------------
 
-  http://www.ginac.de/CLN/
+Please be advised that the following libraries are covered under the GPLv3
```

Comment:
Maybe add a note that from here on all license information concerns libraries that are OPTIONALLY linked in? It does somehow state that in the introductory paragraph, but maybe a section title would make that even more clear?